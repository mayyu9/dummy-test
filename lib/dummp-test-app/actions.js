'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.loadDummpTestApp = loadDummpTestApp;
var LOAD_DUMMP_TEST_APP = exports.LOAD_DUMMP_TEST_APP = 'LOAD_DUMMP_TEST_APP';
var LOAD_DUMMP_TEST_APP_SUCCEEDED = exports.LOAD_DUMMP_TEST_APP_SUCCEEDED = 'LOAD_DUMMP_TEST_APP_SUCCEEDED';
var LOAD_DUMMP_TEST_APP_FAILED = exports.LOAD_DUMMP_TEST_APP_FAILED = 'LOAD_DUMMP_TEST_APP_FAILED';

function loadDummpTestApp(service, patientId) {
  return {
    type: LOAD_DUMMP_TEST_APP,
    data: {
      service: service,
      patientId: patientId
    }
  };
}