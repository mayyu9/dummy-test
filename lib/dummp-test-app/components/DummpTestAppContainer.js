'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sagas = exports.reducers = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _actions = require('../actions');

var _reducers = require('../reducers');

var _reducers2 = _interopRequireDefault(_reducers);

var _sagas = require('../sagas');

var _sagas2 = _interopRequireDefault(_sagas);

var _service = require('../service');

var _service2 = _interopRequireDefault(_service);

var _DummpTestAppView = require('./DummpTestAppView');

var _DummpTestAppView2 = _interopRequireDefault(_DummpTestAppView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  dummpTestApp: _propTypes2.default.string,
  refreshDummpTestApp: _propTypes2.default.func.isRequired,
  isLoading: _propTypes2.default.bool,
  isFailed: _propTypes2.default.bool
};

var DummpTestAppContainer = function (_React$Component) {
  _inherits(DummpTestAppContainer, _React$Component);

  function DummpTestAppContainer() {
    _classCallCheck(this, DummpTestAppContainer);

    return _possibleConstructorReturn(this, (DummpTestAppContainer.__proto__ || Object.getPrototypeOf(DummpTestAppContainer)).apply(this, arguments));
  }

  _createClass(DummpTestAppContainer, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      document.addEventListener('orionReactComponentRefresh', this.props.refreshDummpTestApp);
      this.props.refreshDummpTestApp();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      document.removeEventListener('orionReactComponentRefresh', this.props.refreshDummpTestApp);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          isFailed = _props.isFailed,
          isLoading = _props.isLoading,
          dummpTestApp = _props.dummpTestApp;

      return _react2.default.createElement(_DummpTestAppView2.default, { isFailed: isFailed, isLoading: isLoading, text: dummpTestApp });
    }
  }]);

  return DummpTestAppContainer;
}(_react2.default.Component);

DummpTestAppContainer.propTypes = propTypes;

var mapStateToProps = function mapStateToProps(state) {
  return {
    dummpTestApp: state.dummpTestAppState.dummpTestApp,
    isLoading: state.dummpTestAppState.isLoading,
    isFailed: state.dummpTestAppState.isFailed
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch, ownProps) {
  return {
    refreshDummpTestApp: function refreshDummpTestApp() {
      dispatch((0, _actions.loadDummpTestApp)(ownProps.service || _service2.default, ownProps.patientId));
    }
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(DummpTestAppContainer);


var reducers = _extends({}, { dummpTestAppState: _reducers2.default });
exports.reducers = reducers;


var sagas = _sagas2.default;
exports.sagas = sagas;