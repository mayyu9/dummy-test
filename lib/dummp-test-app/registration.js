'use strict';

var _reactOnRails = require('react-on-rails');

var _reactOnRails2 = _interopRequireDefault(_reactOnRails);

var _DummpTestAppApplication = require('./components/DummpTestAppApplication');

var _DummpTestAppApplication2 = _interopRequireDefault(_DummpTestAppApplication);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_reactOnRails2.default.register({
  DummpTestAppApplication: _DummpTestAppApplication2.default
});