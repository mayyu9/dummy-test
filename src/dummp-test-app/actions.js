export const LOAD_DUMMP_TEST_APP = 'LOAD_DUMMP_TEST_APP';
export const LOAD_DUMMP_TEST_APP_SUCCEEDED = 'LOAD_DUMMP_TEST_APP_SUCCEEDED';
export const LOAD_DUMMP_TEST_APP_FAILED = 'LOAD_DUMMP_TEST_APP_FAILED';

export function loadDummpTestApp(service, patientId) {
  return {
    type: LOAD_DUMMP_TEST_APP,
    data: {
      service,
      patientId,
    },
  };
}
