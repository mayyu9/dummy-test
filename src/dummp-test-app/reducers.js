import {
  LOAD_DUMMP_TEST_APP,
  LOAD_DUMMP_TEST_APP_SUCCEEDED,
  LOAD_DUMMP_TEST_APP_FAILED,
} from './actions';

const dummpTestAppReducers = (state = {
  isFailed: false,
  isLoading: true,
  dummpTestApp: undefined,
}, action) => {
  const newState = Object.assign({}, state);

  switch (action.type) {
    case LOAD_DUMMP_TEST_APP:
      newState.isFailed = false;
      newState.isLoading = true;
      return newState;

    case LOAD_DUMMP_TEST_APP_SUCCEEDED:
      newState.isFailed = false;
      newState.isLoading = false;
      newState.dummpTestApp = action.dummpTestApp;
      return newState;

    case LOAD_DUMMP_TEST_APP_FAILED:
      newState.isFailed = true;
      newState.isLoading = false;
      return newState;

    default:
      return state;
  }
};

export default dummpTestAppReducers;
