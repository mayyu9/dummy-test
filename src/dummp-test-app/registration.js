import ReactOnRails from 'react-on-rails';
import DummpTestAppApplication from './components/DummpTestAppApplication';

ReactOnRails.register({
  DummpTestAppApplication,
});
