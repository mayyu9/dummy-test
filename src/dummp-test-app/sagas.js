import { call, put, takeLatest } from 'redux-saga/effects';

import {
  LOAD_DUMMP_TEST_APP,
  LOAD_DUMMP_TEST_APP_SUCCEEDED,
  LOAD_DUMMP_TEST_APP_FAILED,
} from './actions';

function* fetchDummpTestApp(action) {
  try {
    const dummpTestApp = yield call(action.data.service.getDummpTestApp, action.data.patientId);
    yield put({ type: LOAD_DUMMP_TEST_APP_SUCCEEDED, dummpTestApp });
  } catch (e) {
    yield put({ type: LOAD_DUMMP_TEST_APP_FAILED });
  }
}

function* fetchDummpTestAppSaga() {
  yield takeLatest(LOAD_DUMMP_TEST_APP, fetchDummpTestApp);
}

export { fetchDummpTestApp };
export default [fetchDummpTestAppSaga];
