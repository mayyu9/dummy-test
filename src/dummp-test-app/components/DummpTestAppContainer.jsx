import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import { loadDummpTestApp } from '../actions';
import dummpTestAppReducers from '../reducers';
import dummpTestAppSagas from '../sagas';
import dummpTestAppService from '../service';

import DummpTestAppView from './DummpTestAppView';

const propTypes = {
  dummpTestApp: PropTypes.string,
  refreshDummpTestApp: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  isFailed: PropTypes.bool,
};

class DummpTestAppContainer extends React.Component {
  componentDidMount() {
    document.addEventListener('orionReactComponentRefresh', this.props.refreshDummpTestApp);
    this.props.refreshDummpTestApp();
  }

  componentWillUnmount() {
    document.removeEventListener('orionReactComponentRefresh', this.props.refreshDummpTestApp);
  }

  render() {
    const { isFailed, isLoading, dummpTestApp } = this.props;
    return <DummpTestAppView isFailed={isFailed} isLoading={isLoading} text={dummpTestApp} />;
  }
}

DummpTestAppContainer.propTypes = propTypes;

const mapStateToProps = state => ({
  dummpTestApp: state.dummpTestAppState.dummpTestApp,
  isLoading: state.dummpTestAppState.isLoading,
  isFailed: state.dummpTestAppState.isFailed,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  refreshDummpTestApp: () => {
    dispatch(loadDummpTestApp(ownProps.service || dummpTestAppService, ownProps.patientId));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(DummpTestAppContainer);

const reducers = Object.assign({}, { dummpTestAppState: dummpTestAppReducers });
export { reducers };

const sagas = dummpTestAppSagas;
export { sagas };
