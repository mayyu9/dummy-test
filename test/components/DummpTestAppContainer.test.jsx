/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallowWithStore } from 'enzyme-redux';
import { createMockStore } from 'redux-test-utils';
import DummpTestAppContainer from '../../src/dummp-test-app/components/DummpTestAppContainer';
import { LOAD_DUMMP_TEST_APP } from '../../src/dummp-test-app/actions';

it('should render a DummpTestAppContainer with a populated DummpTestApp', () => {
  const testContainer = shallowWithStore(
    <DummpTestAppContainer />,
    createMockStore({
      dummpTestAppState: {
        dummpTestApp: 'DummpTestApp',
      },
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a DummpTestAppContainer with a nonpopulated DummpTestApp', () => {
  const testContainer = shallowWithStore(
    <DummpTestAppContainer />,
    createMockStore({
      dummpTestAppState: {},
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a DummpTestAppContainer with DummtTestEngine loading', () => {
  const testContainer = shallowWithStore(
    <DummpTestAppContainer />,
    createMockStore({
      dummpTestAppState: {},
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should render a DummpTestApp container with a DummtTestEngine failed', () => {
  const testContainer = shallowWithStore(
    <DummpTestAppContainer />,
    createMockStore({
      dummpTestAppState: {
        isFailed: true,
      },
    }),
  );
  expect(testContainer.first().shallow()).toMatchSnapshot();
});

it('should call the dispatch method', () => {
  const testStore = createMockStore({
    dummpTestAppState: {
    },
  });
  const testContainer = shallowWithStore(
    <DummpTestAppContainer patientId={100} />,
    testStore,
  );

  testContainer.props().refreshDummpTestApp();
  expect(testStore.isActionTypeDispatched(LOAD_DUMMP_TEST_APP)).toBe(true);
});
