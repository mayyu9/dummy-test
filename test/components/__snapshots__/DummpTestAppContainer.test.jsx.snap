exports[`test should render a DummpTestApp container with a DummtTestEngine failed 1`] = `
<InjectIntl(DummpTestAppView)
  isFailed={true} />
`;

exports[`test should render a DummpTestAppContainer with DummtTestEngine loading 1`] = `<InjectIntl(DummpTestAppView) />`;

exports[`test should render a DummpTestAppContainer with a nonpopulated DummpTestApp 1`] = `<InjectIntl(DummpTestAppView) />`;

exports[`test should render a DummpTestAppContainer with a populated DummpTestApp 1`] = `
<InjectIntl(DummpTestAppView)
  text="DummpTestApp" />
`;
